src/kcas.mli

```diff
@@ -1,12 +1,64 @@
 (** {1 Auxiliary modules} *)
 
 module Backoff : module type of Backoff
-(** Randomized exponential backoff mechanism. *)
+module Domain_local_await : module type of Domain_local_await
+
+(** Retry support. *)
+module Retry : sig
+  exception Later
+  (** Exception that may be raised to signal that the operation should be
+      retried, at some point in the future, after the examined shared memory
+      location or locations have changed.
+
+      {b NOTE}: It is important to understand that "{i after}" may effectively
+      mean "{i immediately}", because it may be the case that the examined
+      shared memory locations have already changed. *)
+
+  val later : unit -> 'a
+  (** [later ()] is equivalent to [raise Later]. *)
+
+  val unless : bool -> unit
+  (** [unless condition] is equivalent to [if not condition then later ()]. *)
+end
+
+(** Operating modes of the [k-CAS-n-CMP] algorithm. *)
+module Mode : sig
+  type t
+  (** Type of an operating mode of the [k-CAS-n-CMP] algorithm. *)
+
+  val lock_free : t
+  (** In [lock_free] mode the algorithm makes sure that at least one domain will
+      be able to make progress by performing read-only operations as read-write
+      operations. *)
```

Comment:
I adjust the text.  The cost is actually non-trivial, but generally speaking mutating locations is more expensive unless avoiding the mutation leads to starvation, which is kind of the difference between the modes.  (The `commit` mechanism is designed to avoid starvation due to the obstruction-free mode by switching to the lock-free mode in case interference is detected.)